sdk/program/src/program.rs

```diff
@@ -64,3 +64,42 @@ extern "C" {
         signers_seeds_len: u64,
     ) -> u64;
 }
+
+#[allow(unused)]
+const MAX_RETURN_DATA: usize = 1024;
+
+/// Set return data for program
+pub fn set_return_data(data: &[u8]) {
+    #[cfg(target_arch = "bpf")]
+    {
+        unsafe { sol_set_return_data(data.as_ptr(), data.len() as u64) };
+    }
+
+    #[cfg(not(target_arch = "bpf"))]
+    crate::program_stubs::sol_set_return_data(data)
+}
+
+#[cfg(target_arch = "bpf")]
+extern "C" {
+    fn sol_set_return_data(data: *const u8, length: u64);
+}
+
+/// Get the return data from invoked program
+pub fn get_return_data() -> Vec<u8> {
+    #[cfg(target_arch = "bpf")]
+    {
+        let mut buf = [0u8; MAX_RETURN_DATA];
+
+        let size = unsafe { sol_get_return_data(buf.as_mut_ptr(), buf.len() as u64) };
+
+        buf[..size as usize].to_vec()
+    }
+
+    #[cfg(not(target_arch = "bpf"))]
+    crate::program_stubs::sol_get_return_data()
+}
+
+#[cfg(target_arch = "bpf")]
```

Comment:
These can be placed inside their respective functions, take a look at examples in `Pubkey`

Reply:
Go ahead and move `sol_invoke_signed_rust` in there while you are at it if you want

Reply:
This does look nicer, fixed.